python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -599,11 +614,20 @@ def on_task_output_generated(self, task_index: int, output: RefBundle):
         task_info.bytes_outputs += output_bytes
 
         for block_ref, meta in output.blocks:
-            assert meta.exec_stats and meta.exec_stats.wall_time_s
+            assert (
+                meta.exec_stats is not None and meta.exec_stats.wall_time_s is not None
+            )
             self.block_generation_time += meta.exec_stats.wall_time_s
             assert meta.num_rows is not None
             self.rows_task_outputs_generated += meta.num_rows
             trace_allocation(block_ref, "operator_output")
+            if meta.exec_stats.max_rss_bytes is not None:
```

Comment:
let's make sure it's not None for map ops. 
Either add an assertion here or add a test.